### DISCUSSION TITLE
Recommended setup for local testing of interdependent lambdas

### DISCUSSION THREAD

**Discussion by aochagavia** [cargo-lambda/cargo-lambda#632] [Q&A]
Hello there! First of all thanks for this amazing project. It got me up to speed very quickly, without having used AWS Lambda before.

As to my question, I have a project that involves 3 lambdas (say `parent`, `child1`, `child2`). One of them (`parent`) manually invokes the two children using `aws_sdk_lambda::Client::invoke`. Until now I've been testing it by first deploying `child1` and `child2` to AWS Lambda, and testing `parent` locally using `cargo lambda watch`. However, I'd like to test all the code without having to do a deploy (i.e. I want to tell `parent` to invoke the locally-hosted `child1` and `child2`, without going to the real AWS Lambda).

Is that possible? Otherwise, what would be the recommended way of testing this kind of function?

**Comment by calavera** [comment]
I had never thought of that use case, to be honest. I think the SDK can be configured to send requests to a different host, but I've never looked into it.

**Reply by aochagavia** [reply]
Out of curiosity, how would you tackle local development in this scenario? I'm currently inclined to look into the SDK configuration and try to wire it to cargo lambda, but I'm pretty new to AWS Lambda so if you see any alternatives that would make sense, please tell me :)

**Reply by mxfactorial** [reply]
[AWS_ENDPOINT_URL_LAMBDA](https://docs.aws.amazon.com/sdkref/latest/guide/ss-endpoints-table.html) but its [not yet supported](https://docs.aws.amazon.com/sdkref/latest/guide/feature-ss-endpoints.html#ss-endpoints-sdk-compat) in rust

follow https://github.com/awslabs/aws-sdk-rust/issues/932

---

youre describing microservices

instead of designing lambdas as microservices, consider reducing lambda to a configuration detail:

1. build integrated web services locally without lambda
2. in parent lambda, assign child1, child2 endpoint environment variables as function urls
3. then when youre ready to host your services in lambda, [cargo lambda deploy --layer-arn EXTENSION_ARN](https://www.cargo-lambda.info/guide/lambda-extensions.html#step-4-associate-your-extension-to-a-function) with EXTENSION_ARN assigned to your regions [lambda web adapter layer arn](https://github.com/awslabs/aws-lambda-web-adapter?tab=readme-ov-file#lambda-functions-packaged-as-zip-package-for-aws-managed-runtimes)

lambda web adapter offers many [examples](https://github.com/awslabs/aws-lambda-web-adapter/tree/main/examples)
